Client/src/main/java/com/minecrafttas/tasbattle/Tickratechanger.java

```diff
@@ -0,0 +1,32 @@
+package com.minecrafttas.tasbattle;
+
+import lombok.Getter;
+import net.minecraft.client.Minecraft;
+
+public class Tickratechanger {
+
+	@Getter
+	private float tickrate;
+	
+	@Getter
+	private float gamespeed;
+	
+	@Getter
+	private long msPerTick;
+	
+	public void changeTickrate(float tickrate) {
+		if(tickrate<=0 || tickrate > 1000) return;
+		TASBattle.LOGGER.debug("Changing tickrate to {}", tickrate);
+		
+		this.tickrate=tickrate;
+		this.gamespeed = tickrate/20;
+		this.msPerTick = (long) (1000L/tickrate);
+		
+		changeClientTickrate(msPerTick);
+	}
+	
+	private void changeClientTickrate(long msPerTick) {
+		Minecraft mc = Minecraft.getInstance();
+		mc.timer.msPerTick = msPerTick;
+	}
+}
```

Comment:
tickrate is going to be variable therefore copy the custom tickrate ms hook from lotas tickrate changer. Also you don't have to change the server tickrate ever so ignore that half of the code

Reply:
I don't understand... The tickrate is variable, just without tickadvance and tickrate 0...

Reply:
Problem is changing the tickrate during for example an advancement toast. It would skip weirdly